controllers/machinedeployment_sync.go

```diff
@@ -368,6 +367,23 @@ func calculateStatus(allMSs []*clusterv1.MachineSet, newMS *clusterv1.MachineSet
 		UnavailableReplicas: unavailableReplicas,
 	}
 
+	if *deployment.Spec.Replicas == status.ReadyReplicas {
+		status.Phase = string(clusterv1.MachineDeploymentPhaseRunning)
+	}
+	if *deployment.Spec.Replicas != status.ReadyReplicas && *deployment.Spec.Replicas > status.ReadyReplicas {
+		status.Phase = string(clusterv1.MachineDeploymentPhaseScalingUp)
+	}
+	if totalReplicas-availableReplicas < 0 {
```

Comment:
Can this be merged with the `unavailableReplicas` condition above since `unavailableReplicas = totalReplicas - availableReplicas`.
That way this condition has the benefit of the comment as well.

Reply:
it would have been reset to 0 at this point, didn't want to add another variable, but will put a comment

Reply:
Hmm...it's ok. I don't think that comment on line 376 is necessary. Discard my original comment.